src/python/perfflowaspect/advice_dispatch.py

```diff
@@ -20,17 +29,17 @@ class AdviceDispatcher:
 
     def _dispatch(clobj, pointcut, scope):
         if pointcut == "around":
-            return clobj.around
+            return clobj.around if ChromeTracingAdvice.enable_logging else _disable
         elif pointcut == "before":
-            return clobj.before
+            return clobj.before if ChromeTracingAdvice.enable_logging else _disable
         elif pointcut == "after":
-            return clobj.after
+            return clobj.after if ChromeTracingAdvice.enable_logging else _disable
         elif pointcut == "around_async":
-            return clobj.around_async
+            return clobj.around_async if ChromeTracingAdvice.enable_logging else _disable
         elif pointcut == "before_async":
-            return clobj.before_async(scope)
+            return clobj.before_async(scope) if ChromeTracingAdvice.enable_logging else _disable
         elif pointcut == "after_async":
-            return clobj.after_async(scope)
+            return clobj.after_async(scope) if ChromeTracingAdvice.enable_logging else _disable
```

Comment:
I believe this only addresses the `critical_path` enable/disable which is called from [here](https://github.com/flux-framework/PerfFlowAspect/blob/main/src/python/perfflowaspect/aspect.py#L26), as the `get` method calls the `_dispatch`. 
Assuming I understood the flow correctly (and I may be missing something), the [`sync_event` ](https://github.com/flux-framework/PerfFlowAspect/blob/main/src/python/perfflowaspect/aspect.py#L36) and [`async_event`](https://github.com/flux-framework/PerfFlowAspect/blob/main/src/python/perfflowaspect/aspect.py#L41) call `get_advice`, followed by `clobj.sync_event` or `clobj.async_event` -- which doesn't go through the `_dispatch` function above.  

I am wondering if this logic needs to be in `aspect.py` instead of `advice_dispatch.py`?

Reply:
@tpatki Good catch, yes there were cases that were missed by putting the logic in `advice_dispatch.py`. I have moved the logic into `aspect.py` and added more python unit tests to test this case.